data-prepper-core/src/main/java/org/opensearch/dataprepper/core/livecapture/LiveCaptureManager.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.core.livecapture;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.EventMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.RateLimiter;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Manages live capture functionality for Data Prepper events.
+ * Coordinates event marking, rate limiting, and capture entry management.
+ */
+public class LiveCaptureManager {
+    private static final Logger LOG = LoggerFactory.getLogger(LiveCaptureManager.class);
+    public static final String LIVE_CAPTURE_FLAG = "liveCapture";
+    public static final String LIVE_CAPTURE_OUTPUT = "liveCaptureOutput";
+    public static final String SOURCE = "Source";
+    public static final String PROCESSOR = "Processor";
+    public static final String ROUTE = "Route";
+    public static final String SINK = "Sink";
+    private static volatile LiveCaptureManager INSTANCE;
```

Comment:
I am not sure if we need this.  For now, we can keep it but we should not need this.